core/services/relay/evm/functions/threshold_config_poller.go

```diff
@@ -0,0 +1,167 @@
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/pg"
+)
+
+type ThresholdConfigPoller interface {
+	ocrtypes.ContractConfigTracker
+
+	Replay(ctx context.Context, fromBlock int64) error
+}
+
+// ConfigSet Common to all OCR2 evm based contracts: https://github.com/smartcontractkit/libocr/blob/master/contract2/dev/OCR2Abstract.sol
+var ConfigSet common.Hash
+
+var defaultABI abi.ABI
+
+const configSetEventName = "ConfigSet"
+
+func init() {
+	var err error
+	abiPointer, err := ocr2aggregator.OCR2AggregatorMetaData.GetAbi()
+	if err != nil {
+		panic(err)
+	}
+	defaultABI = *abiPointer
+	ConfigSet = defaultABI.Events[configSetEventName].ID
+}
+
+func unpackLogData(d []byte) (*ocr2aggregator.OCR2AggregatorConfigSet, error) {
+	unpacked := new(ocr2aggregator.OCR2AggregatorConfigSet)
+	err := defaultABI.UnpackIntoInterface(unpacked, configSetEventName, d)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to unpack log data")
+	}
+	return unpacked, nil
+}
+
+func configFromLog(logData []byte) (ocrtypes.ContractConfig, error) {
+	unpacked, err := unpackLogData(logData)
+	if err != nil {
+		return ocrtypes.ContractConfig{}, err
+	}
+
+	var transmitAccounts []ocrtypes.Account
+	for _, addr := range unpacked.Transmitters {
+		transmitAccounts = append(transmitAccounts, ocrtypes.Account(addr.String()))
+	}
+	var signers []ocrtypes.OnchainPublicKey
+	for _, addr := range unpacked.Signers {
+		addr := addr
+		signers = append(signers, addr[:])
+	}
+
+	thresholdOffchainConfig, err := DecodeThresholdReportingPluginConfig(unpacked.OffchainConfig)
+	if err != nil {
+		return ocrtypes.ContractConfig{}, errors.Wrapf(err, "failed to decode Threshold plugin offchain config: %v", unpacked.OffchainConfig)
+	}
+
+	thresholdOffchainConfigBytes, err := EncodeThresholdPluginConfig(thresholdOffchainConfig)
+	if err != nil {
+		return ocrtypes.ContractConfig{}, errors.Wrapf(err, "failed to encode Threshold plugin offchain config: %v", unpacked.OffchainConfig)
+	}
+
+	return ocrtypes.ContractConfig{
+		ConfigDigest:          unpacked.ConfigDigest,
+		ConfigCount:           unpacked.ConfigCount,
+		Signers:               signers,
+		Transmitters:          transmitAccounts,
+		F:                     unpacked.F,
+		OnchainConfig:         unpacked.OnchainConfig,
+		OffchainConfigVersion: unpacked.OffchainConfigVersion,
+		OffchainConfig:        thresholdOffchainConfigBytes,
+	}, nil
+}
+
+type configPoller struct {
+	lggr               logger.Logger
+	filterName         string
+	destChainLogPoller logpoller.LogPoller
+	addr               common.Address
+}
+
+func configPollerFilterName(addr common.Address) string {
+	return logpoller.FilterName("OCR2ConfigPoller", addr.String())
+}
+
+// NewConfigPoller creates a new ConfigPoller
+func NewThresholdConfigPoller(lggr logger.Logger, destChainPoller logpoller.LogPoller, addr common.Address) (ThresholdConfigPoller, error) {
```

Comment:
Why not NewFunctionsConfigPoller(target_config, ...) ?

Reply:
We don't want a custom config poller for Functions.  We only want a custom config poller for Threshold & S4, right?

Reply:
I'm suggesting to have one custom poller to support all 3. If we're already creating a custom one (which obviously depends on Functions config) why not have it support all our plugins?

Reply:
I do not think that is possible.  You need to have a separate instance of the config poller for each plugin.  I do not know how to implement a single poller that implements the logic for each plugin.  The ConfigPoller has a particular interface to which we must adhere.  We cannot add an additional flag parameter to any of these functions without violating the interface.

Reply:
Let me try more before I confirm this won't work though.